src/framework/data/createIntegrationRelationship.ts

```diff
@@ -252,9 +256,10 @@ export function generateRelationshipKey(
   to: { _key: string } | string,
 ): string {
   if (!from || !to) {
-    throw new Error(
-      '"from" and "to" must be provided to generate a relationship _key!',
-    );
+    throw new IntegrationError({
+      code: 'GENERATE_RELATIONSHIP_KEY_MISSING_RELATIONSHIP_FROM_OR_TO',
+      message: '"from" and "to" must be provided to generate a relationship _type!',
+    });
```

Comment:
I've preferred something along the lines of `throw new IntegrationRelationshipTypeGenerationError(from, to)`, and putting the code and message generation in the error constructor to avoid duplication and to construct a message with more detail, to avoid "this could be the problem or that could be the problem". I see other errors doing some of this, such as `IntegrationLocalConfigFieldMissingError`, by putting the `code` in the constructor 👍